tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPSyncMirrorTest.java

```diff
@@ -0,0 +1,481 @@
+         Session session = connection.createSession(transactional, transactional ? Session.SESSION_TRANSACTED : Session.CLIENT_ACKNOWLEDGE);
+         MessageProducer producer = session.createProducer(session.createQueue(getQueueName()));
+
+         connection.start();
+
+         producer.setDeliveryMode(DeliveryMode.PERSISTENT);
+
+         final String bodyMessage;
+         {
+            StringBuffer buffer = new StringBuffer();
+            for (int i = 0; i < messageSize; i++) {
+               buffer.append("large Buffer...");
+            }
+            bodyMessage = buffer.toString();
+         }
+
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            if (logger.isTraceEnabled()) {
+               logger.trace("===>>> send message {}", i);
+            }
+            int theI = i;
+            sendPending.countUp();
+            logger.trace("semSend.acquire");
+            semSend.acquire();
+            if (!transactional) {
+               pool.execute(() -> {
+                  try {
+                     if (logger.isTraceEnabled()) {
+                        logger.trace("Entering non TX send with sendPending = " + sendPending.getCount());
+                     }
+                     TextMessage message = session.createTextMessage(bodyMessage);
+                     message.setStringProperty("strProperty", "" + theI);
+                     producer.send(message);
+                     sendPending.countDown();
+                     if (logger.isTraceEnabled()) {
+                        logger.trace("leaving non TX send with sendPending = " + sendPending.getCount());
+                     }
+                  } catch (Throwable e) {
+                     logger.warn(e.getMessage(), e);
+                     errors.incrementAndGet();
+                  }
+               });
+            } else {
+               CountDownLatch sendDone = new CountDownLatch(1);
+               pool.execute(() -> {
+                  try {
+                     TextMessage message = session.createTextMessage(bodyMessage);
+                     message.setStringProperty("strProperty", "" + theI);
+                     producer.send(message);
+                  } catch (Throwable e) {
+                     errors.incrementAndGet();
+                     logger.warn(e.getMessage(), e);
+                  }
+                  sendDone.countDown();
+               });
+
+               Wait.assertEquals(i, replicatedQueue::getMessageCount);
+
+               Assert.assertTrue(sendDone.await(10, TimeUnit.SECONDS));
+
+               pool.execute(() -> {
+                  try {
+                     session.commit();
+                     sendPending.countDown();
+                  } catch (Throwable e) {
+                     e.printStackTrace();
+                  }
+               });
+            }
+
+            Assert.assertFalse("Send epending not supposed to succeed", sendPending.await(10, TimeUnit.MILLISECONDS));
```

Comment:
probably meant: ""Send pending not supposed to succeed"

Reply:
thanks!